src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ScheduleResourceIntTest.kt

```diff
@@ -46,16 +49,15 @@ class ScheduleResourceIntTest : IntegrationTestBase() {
 
     @BeforeEach
     fun setUp() {
-      testData(repository) {
+      nomisDataBuilder.build {
         programService {
           courseActivity = courseActivity {
             payRate()
-            courseSchedule(scheduleDate = today.toString())
+            courseSchedule = courseSchedule(scheduleDate = today.toString())
             courseScheduleRule()
           }
         }
       }
-      courseSchedule = courseActivity.courseSchedules.first()
     }
```

Comment:
An example of using an entity generated by the DSL in a test - we keep a copy of the `courseActivity` and the `courseSchedule`.
As every DSL function returns an entity we could also do this for `payRate()`  or `courseScheduleRule()`.